### PR TITLE
Panel component - new variant sizes that adjust across breakpoints

### PR DESCRIPTION
**Pull request by philworthington**
This updates the Panel component with variant classes that adjust the panel sizes across all breakpoints. Sass file cleaned up. Variant template files cleaned up and updated.

### CODE REVIEW

**Comment by philworthington** [comment]
Polish updates from design need to be made, hold off on review

File: src/components/Panel/Panel.ExtraLarge.html [hunk 1]
```diff
@@ -1,116 +1,44 @@
...
+<button class="ms-Button ms-PanelAction-open">
   <span class="ms-Button-label">Open panel</span>
 </button>
 
-<div class="ms-Panel ms-Panel--extraLarge">
-  <div class="ms-Overlay ms-Overlay--dark js-togglePanel"></div>
+<div class="ms-Panel ms-Panel--xl">
+  <div class="ms-Overlay ms-PanelAction-close"></div>
   <div class="ms-Panel-main">
-    <div class="ms-Panel-commands">
```
Comment (battletoilet) [inline src/components/Panel/Panel.ExtraLarge.html]: Is there a reason for removing Command Bar?

**Comment by battletoilet** [comment]
This looks pretty good, i tested it and it works great. Not sure if there is supposed to be content inside of the smallest panel. Also i left a comment about command bar but otherwise I LIKE WHAT YOU GOT.

File: src/components/Panel/Panel.scss [hunk 2]
```diff
@@ -39,153 +46,103 @@ $CommandButtonBlue: #2488D8;
...
-  top: 40px; // Push it below the commands.
-  bottom: 0;
-  left: 0;
-  right: 0;
-  padding: 0 32px 20px 32px;
-  overflow-y: auto;
-}
-
-// Header text at the top of the panel.
-.ms-Panel-headerText {
```
Comment (mikewheaton) [inline src/components/Panel/Panel.scss]: What can we do about panels with long header text? See issue #320 

If we're limiting it to one line, we should probably truncate with ellipses at the end.

**Comment by mikewheaton** [comment]
`.ms-Panel-commands` and `.ms-Panel-contentInner` are set to `display: none;` in the "small, left-aligned, fixed" and "small, light dismiss, fixed" panels, so no content is visible.

**Comment by notandrew** [comment]
Please see https://github.com/OfficeDev/Office-UI-Fabric/issues/320

**Comment by philworthington** [comment]
@mikewheaton The small left aligned and small light dismiss panels are only to be used for Left Nav and the contextual menu on mobile respectively. The LeftNav or contextual menu would replace the contentInner and there would be no commands for this panel. This isn't apparent in the templates right now because we don't have a LeftNav component and any contextual menu updates should be a separate item.

Unfortunately we don't have a better way to view the panel in the sample page right now - using the buttons on the page to change the variants is just adding/removing classes in the default panel template, not actually using a different template and markup for each variant altogether. 

The important part was getting the variant size classes correct across breakpoints and making them available so that they could be consumed in other experiences. We should make updates to these panel templates when a leftnav and contextual menu are ready.

**Comment by mikewheaton** [comment]
Sounds good, approved!

**Pull request merged**